src/routes/category.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{
+    models::{
+        category::{Category, CreateCategory},
+        error::Error,
+        response::{Empty, Response},
+        UserRecordId,
+    },
+    utils::{category, session},
+    Result,
+};
+use rocket::{post, serde::json::Json, tokio::fs::remove_dir_all, State};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use surrealdb::{engine::remote::ws::Client, Surreal};
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "rocket::serde")]
+pub struct CategoryData<'r> {
+    pub id: &'r str,
+    pub token: &'r str,
+
+    pub data: CreateCategory,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateCatResponse {
+    pub id: String,
+}
+
+#[post("/create", data = "<category>")]
+pub async fn create(
+    db: &State<Surreal<Client>>,
+    category: Json<CategoryData<'_>>,
+) -> Result<CreateCatResponse> {
+    if !session::verify(db, category.id, category.token).await {
+        return Err(Error::Unauthorized(Json(
+            "Failed to grant permission".into(),
+        )));
+    }
+
+    let data = category::create(db, category.into_inner().data)
+        .await
+        .map_err(|e| Error::ServerError(Json(e.to_string().into())))?
+        .ok_or(Error::ServerError(Json("Failed to create category".into())))?;
+
+    Ok(Json(Response {
+        success: true,
+        message: "Category created successfully".into(),
+        data: Some(CreateCatResponse {
+            id: data.id.unwrap().id.to_string(),
+        }),
+    }))
+}
+
+#[post("/delete/<id>", data = "<category>")]
+pub async fn delete(
+    db: &State<Surreal<Client>>,
+    id: &str,
+    category: Json<CategoryData<'_>>,
+) -> Result<Empty> {
+    if !session::verify(db, category.id, category.token).await {
+        return Err(Error::Unauthorized(Json(
+            "Failed to grant permission".into(),
+        )));
+    }
+
+    category::delete(db, id)
+        .await
+        .map_err(|e| Error::ServerError(Json(e.to_string().into())))?;
+
+    remove_dir_all(Path::new("content/").join(id))
```

Comment:
Are you sure? We should not delete all user contents if we just remove a category!